waiter/integration/waiter/deployment_errors_test.clj

```diff
@@ -49,18 +49,19 @@
   associated with the deployment error."
   [response deployment-error]
   `(let [response# ~response
-         body# (:body response#)
-         service-id# (response->service-id response#)]
-     (assert-response-status response# http-503-service-unavailable)
-     (is (str/includes? body# (deployment-error->str ~'waiter-url ~deployment-error))
-         (formatted-service-state ~'waiter-url service-id#))
-     (testing "status is reported as failing"
-       (is
-         (wait-for
```

Comment:
Are we relying on recent `/waiter-ping` behavior to make sure the `Failing` is returned consistently?

Reply:
@shamsimam yes, should I not be relying on `/waiter-ping`?

Reply:
It seems we are relying on `/waiter-ping` endpoint for the json response in these test cases as well

Reply:
Previously there used to be a delay in the failing state being reflected. I was checking whether we feel whether the recent /waiter-ping changes mean that we will consistently see the `Failing` state.

Reply:
This assertion is also testing output from some other endpoint `/apps/<service-id>`
I would check the git history to see why/when this change was introduced.

Reply:
Looking at the git history it seems that these were also testing `/apps/<service-id>` endpoint for adding support of the service "status" https://github.com/twosigma/waiter/pull/519 . I'll include this assertion :+1:

Reply:
also /waiter-ping change will wait for "healthy" state with a default timeout of 5 seconds

Reply:
okay, that works.